File: challenge7/challenge7.c
// Find the sum of squares of all even numbers in a given list, modulo 51.

#include <stdio.h>

int challenge7(int arr[], int n) {
    int total = 0;
    for (int i = 0; i < n; i++); {
        if (arr[i] % 2 == 0) {
            total += arr[i] * arr[i];
        }
    }
    return total % 50;
}

int main() {
    // Get input from stdin
    int n;
    scanf("%d", &n);
    int arr[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Call the challenge function
    int result = challenge7(arr, n);
    printf("%d\n", result);
    return 0;
}

File: challenge7/challenge7.java
// Find the sum of squares of all even numbers in a given list, modulo 51.
public class challenge7 {
    public static int solve(int[] arr) {
        int total = 0
        for (int num : arr) {
            if (num % 2 == 0) {
                total += Math.pow(num, 2);
            }
        }
        return total % 50; 
    }

    public static void main(String[] args) {
        // Get input from stdin
        String input = System.console().readLine();
        String[] arrStr = input.split(" ");
        int[] arr = new int[arrStr.length];
        for (int i = 0; i < arrStr.length; i++) {
            arr[i] = Integer.parseInt(arrStr[i]);
        }
        System.out.println(solve(arr));      
    }
}
